codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/CodegenVisitor.java

```diff
@@ -244,11 +244,11 @@ void execute() {
 
             writers.useFileWriter("endpoints.go", settings.getModuleName(), writer -> {
                 ProtocolGenerator.GenerationContext context = contextBuilder.writer(writer).build();
-                protocolGenerator.generateEndpointRulesEngine(context);
+                protocolGenerator.generateEndpointResolutionV2(context);
```

Comment:
nit: You probably don't need to include the `v2` in the codegen component naming, it could simply be `generateEndpointResolver`